g3w-admin/qdjango/utils/models.py

```diff
@@ -163,4 +163,19 @@ def get_view_layer_ids(user, project):
             set([l.qgs_layer_id for l in get_objects_for_user(get_anonymous_user(), 'qdjango.view_layer', Layer).
                 filter(project=project)])
         )
-    )
\ No newline at end of file
+    )
+
+def get_geocoding_providers():
+    """
+    Return al tuple for qdjango project form field geocoding_providers
+
+    :return: tuple of items for a django form select
+    """
+
+    ret = []
+    for gp, p in settings.GEOCONDING_PROVIDERS.items():
+        if gp == 'bing' and 'bing' not in settings.VENDOR_KEYS:
+            continue
```

Comment:
@wlorenzetti hiding it can become difficult to use (and discover), I would recommend showing an informational message instead, eg:

![image](https://github.com/g3w-suite/g3w-admin/assets/9614886/8bcfa3a4-87a8-4421-b0de-e4841bdaf417)

```html
<div class="help-block" style="color: red">
  ⚠️ No <a href="https://g3w-suite.readthedocs.io/en/latest/settings.html#vendors-settings" target="_blank"><code>VENDOR_KEYS</code></a> found for "Bing" provider
</div>
```

Correct me if I'm wrong, only `Nominatim` provider will be activated by default for each project (so it wouldn't send invalid provider configurations to the client anyway):

![image](https://github.com/g3w-suite/g3w-admin/assets/9614886/0f5dcf04-54af-45b6-b581-f5e52133a17e)

Reply:
@Raruto for me is correct not show Bing if a a vendor key for it is not set inside the settings. It could be useful add a description for the find int he form writing a thing like this: "For use Bing geoding is necessary have a Bing API key "

> Correct me if I'm wrong, only Nominatim provider will be activated by default for each project (so it wouldn't send invalid provider configurations to the client anyway):

No why? at most you will have a map control without provider in the configuration API

Reply:
> for me is correct not show Bing if a a vendor key for it is not set inside the settings

I said this because it complicates development a bit.

API keys (and other server configurations) can also be added on the fly via JS.

If we also have to modify the docker configuration it becomes an extra step (a "duplicated" step) which would require a dedicated documentation..

Even on the server side, a sysadmin might even be happy to simply do that:

```py
GEOCONDING_PROVIDERS = {
  "bing": {
    "label": "Bing",
    "url": "https://dev.virtualearth.net/REST/v1/LocalSearch/?key=my.super.secret.key"
  }
}
```

which is equivalent to the wordy version:

```py
VENDOR_KEYS = {
 'bing': 'my.super.secret.key'
}

GEOCONDING_PROVIDERS = {
  "bing": {
    "label": "Bing",
    "url": "https://dev.virtualearth.net/REST/v1/LocalSearch/"
  }
}
```

But without the risk to lose them in configuration files.

In essence, let's leave people a little more freedom on how to handle it.

Reply:
@Raruto ok I can understand what you are saying me, but the VENDORS_KEYS are used also for other features i.e. base layers .. so I think if better leave one place where to set the VENDOR key.